Level 2

yixwellet opened the pull request.

yixwellet committed:
Отредактирован README

yixwellet committed:
Выполнена первоначальная настройка проекта

1. созданы вспомогательные файлы
2. инициализирован package.json
3. написаны скрипты для запуска проекта в режимах разработки и продакшн
4. установлены необходимые пакеты

yixwellet committed:
Выполнена настройка Webpack

1. настроен webpack.config.js
2. настроен Бабель
3. Настроен PostCSS
4. написаны скрипты для финальной сборки, для разработки и для деплоя на gh-pages

yixwellet committed:
Настройки и общая структура главной страницы:

1. решена проблема с зависимостями (не устанавливался imagemin-gifsicle, из-за чего пришлось откатиться на более старые версии пакетов)
2. настроено кэширование файлов
3. написана общая структура html-файла

yixwellet committed:
Сверстан header для наибольшего разрешения

yixwellet committed:
Сверстана секция lead с заголовком сайта и строкой поиска для наибольшего разрешения

yixwellet committed:
Сверстаны секции about и footer для наибольшего разрешения

yixwellet committed:
Header сделан адаптивным.
Добавлен js модуль, открывающий и закрывающий меню на мобильных устройствах

yixwellet committed:
Ширина меню для мобильных устройств ограничена 320px

yixwellet committed:
Секция lead сделана адаптивной

yixwellet committed:
Секции результатов поиска, информации о разработчике и футер сделаны адаптивными

yixwellet committed:
Сверстана вторая страница, реализована адаптивность

yixwellet committed:
Сверстаны модальные окна

yixwellet committed:
Мелкие правки хедера

yixwellet committed:
Правка README

yixwellet committed:
Правка README

yixwellet committed:
Установлен пакет gh-pages

yixwellet committed:
Изменены настройки webpack.config

coolswood commented:
Удалять папку dist необходимо средствами webpack, а не прописывать удаление в скрипте.

coolswood commented:
# Резюме по работе:
## Инфраструктура и Git

### Критерии, влияющие на работоспособность

- [x] <!--8--> Вебпак собирает проект: кроме кода в сборку должны попасть SVG-изображения и шрифты; 
- [x] <!--8--> Проект выложен на GH-pages
- [x] <!--8--> Продакшн-бандл должен деплоиться на Github Pages скриптом вебпака.
- [x] <!--8--> Устранена проблема с нижними подчеркиваниями при публикации на github. Эта задача решается размещением в корне пустого файла .nojekyll. Как это делать, описано в одной из проектных работ: https://bit.ly/2W4Okyt. 

### Хорошие практики
- [x] <!--6.4--> Переменные окружения запрещены, проект собирается и запускается на любой OC после установки необходимых npm-зависимостей — то есть проект собирается, если клонировать репозиторий на любой компьютер и ввести команды npm i и npm run dev;
- [x] <!--6.4--> Локальный сервер установлен и настроен.
- [x] <!--6.4--> Хеши проставляются и настроено автоматическое обновление страницы (*hot reload*). 
- [x] <!--6.4--> Настроен Babel
- [x] <!--4.8--> PostCSS установлен и настроен для минификации CSS-кода и простановки вендорных префиксов. 
- [x] <!--6.4--> Работа с Git
- [x] <!--6.4--> Исходники проекта хранятся в каталоге `src/`.
- [x] <!--4.8--> В .gitignore есть node_modules

### Рекомендации
- [ ] <!--4--> Для каждой страницы скомпилирован свой CSS-бандл 
- [ ] <!--4--> Бандлы хранятся в отдельной директории. Нельзя сложить их в родительском каталоге dist/
- [ ] <!--4--> Полно и корректно оформлен README.md. По ссылке наши рекомендации: https://code.s3.yandex.net/frontend-developer/landings/cover-letter/index.html. 
- [ ] <!--4--> .gitignore содержит все необходимые каталоги и файлы
- [ ] <!--4--> У каждого установленного плагина зафиксирована версия.

## Количество баллов: 


## HTML и CSS
### На тройку


- [ ] <!--14.429--> Проект адаптирован под различные разрешения экрана, соответствует макетам, сделанным для них. Горизонтальный скролл не должен возникать на разрешениях от 320 пикселей и больше. Скрывать полосу прокрутки свойством `overflow: hidden` нельзя.
- [ ] <!--14.429--> Сайт состоит из 2-х страниц, корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. 
- [ ] <!--14.429--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [ ] <!--9.179--> Отсутствуют ошибки валидации 
- [ ] <!--9.179--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [ ] <!--9.179--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [ ] <!--9.179--> Каркас макета реализован на Flex layout или Grid layout; 

### На четвёрку
- [ ] <!--2.143--> Именование классов и структура файлов сделаны по БЭМ. 
- [ ] <!--2.143--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. Если кнопка в макете должна менять цвет при наведении — это обязательно реализовать в проекте. 
- [ ] <!--2.143--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [ ] <!--2.143--> Шрифты подключены через @font-face.
- [ ] <!--2.143--> reset.css запрещен;
- [ ] <!--2.143--> Элементы формы должны выделяться, когда на них установлен фокус.
- [ ] <!--2.143--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### На пятёрку
- [ ] <!--1.250--> Для изображений задан атрибут alt с подходящий значением.
- [ ] <!--1.250--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [ ] <!--1.250--> растровые и векторные изображения оптимизированы;
- [ ] <!--1.250--> Использование современных HTML-тегов и атрибутов для респонсив изображений, такие как picture для изображений.

## Количество баллов:

yixwellet committed:
Зафиксированы версии плагинов
В скрипте build команда rm заменена на rimraf
JS-файлы перенесены в папку script и распределены по директориям
